src/Security/Jwks/Fetcher/JwksFetcher.php

```diff
@@ -74,4 +81,39 @@ public function fetchKey(string $jwksUrl, string $kId): Key
 
         throw new LtiException(sprintf('Could not find key id %s from url %s', $kId, $jwksUrl));
     }
+
+    /**
+     * @throws GuzzleException
+     * @throws InvalidArgumentException
+     */
+    private function fetchJwksData(string $jwksUrl): array
+    {
+        $cacheKey = sprintf('%s-%s', self::CACHE_PREFIX, base64_encode($jwksUrl));
+
+        if ($this->cache) {
+            $item = $this->cache->getItem($cacheKey);
```

Comment:
Can you protect this by trying to catch to avoid single point of failure if, for example, we have a cache or network issues?

Same everywhere when the cache is used.

Thanks

Reply:
it is already.

this is a private function, called in the public `fetchKey()` one.
(where I `} catch (Throwable|CacheException $exception) {`)

Reply:
As discussed, this solution is still a single point of failure if the cache is down.

we should still try to do the HTTP call to have the result.

I know that then the API receiving the HTTP calls will not be protected by the cache layer and then be under stress but that's where the auto-scaling should help or pre-scaling.

What matters the most is that all those faults will be hidden for the user and the functionality will still work.

Reply:
provided the following:
- possibility to provide a PSR3 logger at construction (if not, auto use the NullLogger)
- it will try to fetch from cache first, log if any cache read issue
- if not found form cache, it will http call the JWKS url
- it'll then try to save into cache if prior http call, and log any cache write issue

=> this way, if the cache is down:
- we fallback to JWKS http call on the owner servers
- and we log as error level that the cache is KO to minimise the potential massive amount of requests we may send

Reply:
Perfect thanks for taking the time to fix this 👍